modules/accumulo/src/main/java/io/fluo/accumulo/iterators/TimestampSkippingIterator.java

```diff
@@ -103,7 +103,13 @@ public void skipColumn(Key curCol) throws IOException {
         && curCol.equals(source.getTopKey(), PartialKey.ROW_COLFAM_COLQUAL_COLVIS)) {
       if (count == 10) {
         Key seekKey = curCol.followingKey(PartialKey.ROW_COLFAM_COLQUAL_COLVIS);
-        Range newRange = new Range(seekKey, true, range.getEndKey(), range.isEndKeyInclusive());
+        Range newRange;
+        if (range.afterEndKey(seekKey)) {
+          newRange =
+              new Range(range.getEndKey(), true, range.getEndKey(), range.isEndKeyInclusive());
```

Comment:
I added a comment to the code explaining the purpose.  Also the code was slightly incorrect.  

I wrote the following little test and it prints false three times.   I had to change the end key's inclusiveness to false to make sure it always prints false.

```java
    Key k0 = new Key("r1","fam1","q1",56437l);
    Key k1 = new Key("r1","fam1","q1",56436l);
    Key k2 = new Key("r1","fam1","q1",56435l);

    Range r1 = new Range(k1, true, k1, false);

    System.out.println(r1.contains(k0));
    System.out.println(r1.contains(k1));
    System.out.println(r1.contains(k2));
```